core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -606,22 +641,51 @@ private void maybeCompleteInitializationWithException(
             return;
         }
 
+        // Remove the partition from the cache as it's failed to initialize.
+        partitionCacheMap.remove(sharePartitionKey);
+        // The partition initialization failed, so complete the request with the exception.
+        // The server should not be in this state, so log the error on broker and surface the same
+        // to the client. The broker should not be in this state, investigate the root cause of the error.
+        log.error("Error initializing share partition with key {}", sharePartitionKey, throwable);
+        maybeCompleteShareFetchExceptionally(future, Collections.singletonList(topicIdPartition), throwable);
+    }
+
+    private void handleSharePartitionException(
```

Comment:
I suggest "fenced" should appear in the method name. This is essentially seeing if the exception indicates fencing and then discarding the share partition from the cache.

Reply:
My idea was to have general exception handling method as we might find other exceptions which needs some handling as well. Do you think we should strictly have fenced special handling method?

Reply:
My point was that you're really only handling a couple of exceptions at the moment, and they're both fencing-related. Anyway, just a suggestion.

Reply:
Thanks, I have changed the method name. Might move to generic one if required in future.